src/core/dom/layout/viewport-observer.js

```diff
@@ -39,10 +46,27 @@ const viewportCallbacks = new WeakMap();
  *
  * @param {Element} element
  * @param {function(IntersectionObserverEntry):void} callback
+ * @param {IOOptions} [opts]
  * @return {import('#core/types/function/types').UnlistenCallback} clean up closure to unobserve the element
  */
-export function observeIntersections(element, callback) {
+export function observeIntersections(element, callback, opts) {
   const win = getWin(element);
+
+  if (opts) {
+    // If there are opts, the IntersectionObserver isn't reusable
+    const viewportObserverNoCache = createViewportObserver(
+      (entries) => {
+        callback(entries[0]);
```

Comment:
Use the last entry, since the `isIntersecting` can change if multiple entries are batched.
```suggestion
        callback(entries[entries.length - 1]);
```

Reply:
Since this `IntersectionObserver` is not cached or reused (due to the presence of `options`), it only observes a single element (L64) so there will only ever be 1 entry.

Reply:
This is incorrect. The entries are batched into the event queue, and an element that enters and exists the viewport. Eg, if a mutation were to happen sync that caused the bounding box to grow and shrink, or a CLS that adjusted the in-viewport elements. Specifically, these are macro tasks and not micro tasks, so multiple for the same element could be queued.

Reply:
Oh, I see.  I didn't know about this behavior.  Updating to use `.length - 1` makes sense.

Reply:
I'm interested to know more about this queueing behavior, but I'm having trouble finding any good references ... can you help me find a good reference?

The only thing I can find is MDN which is kinda ambiguous (no mention what happens if a target has multiple changes):
> The list of entries received by the callback includes one entry for each target which reported a change in its intersection status. Check the value of the [isIntersecting](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserverEntry/isIntersecting) property to see if the entry represents an element that currently intersects with the root.

Reply:
I can't seem to generate a case that causes it, but it falls out of the current specifications tracking of thresholds. Essentially:

> If thresholdIndex does not equal previousThresholdIndex or if isIntersecting does not equal previousIsIntersecting, [queue an IntersectionObserverEntry](https://w3c.github.io/IntersectionObserver/#queue-an-intersectionobserverentry), passing in observer, time, rootBounds, targetRect, intersectionRect, isIntersecting, and target.
- https://w3c.github.io/IntersectionObserver/#update-intersection-observations-algo

This is the only deduplication logic done in the spec, so if we can trigger multiple threshold changes (which can be just 0 -> nonzero -> 0 -> nonzero), then we can queue multiple entries for a target. But this is dependent on when the browser drains the InOb task source queue, which is "implementation defined" in https://html.spec.whatwg.org/multipage/webappapis.html#event-loop-processing-model.

It could be that all current implementations drain the queue before the next time a rendering step can happen.